chart/templates/serviceaccount.yaml

```diff
@@ -1,30 +1,55 @@
+  {{- if .Values.serviceAccount.create -}}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ include "keptn-service.serviceAccountName" . }}
+  name: {{ include "unleash-service.serviceAccountName" . }}
   labels:
-    {{- include "keptn-service.labels" . | nindent 4 }}
+    {{- include "unleash-service.labels" . | nindent 4 }}
   {{- with .Values.serviceAccount.annotations }}
   annotations:
-  {{- toYaml . | nindent 4 }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
-
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: unleash-service
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - list
+      - update
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - create
+      - update
+      - patch
```

Comment:
May I ask for a short description/reasoning why unleash-service needs access to
* pods
* secrets
* events

Reply:
Adding @Raffy23 and/or @pchila to this, maybe they also have some ideas :)

Reply:
After looking at the code again, it seems that the service only needs secret permissions. I somehow thought that the service interacts with pods when toggling a feature, but the code suggests otherwise.